### PR TITLE
Allow overriding NOCASE correlation with Latin1 for iQacs stats collection for duplicate codevalues

### PR DESCRIPTION
**Pull request by khanaffan**
This PR DOES NOT enable ICU support in SQLite. It does not change the default NOCASE collation.

PR changes will be used to analyze existing iModels to see if they have Latin1 case folding issues in iQacs.

### CODE REVIEW

**Review by nick4598 — APPROVED** [review]

**Review by aruniverse — CHANGES_REQUESTED** [review]
isnt this breaking/write incompatible because it changes the index generated?

**Comment by khanaffan** [comment]
> Isn't this breaking/write incompatible because it changes the index generated?

No, as the comment in PR says it does not change any default behavior. We are merging this so iQacs can find out stats about how many iModels in PROD has this issue. It will help us plan how we proceed/release it for all. Though i think the minimum prerequisite will be CodeService. That service has to have this feature on and at some point when everyone uses that service we can enable it by default.

**Review by ColinKerr — APPROVED** [review]

**Review by aruniverse — APPROVED** [review]

File: .vscode/launch.json [hunk 1]
```diff
@@ -53,7 +53,7 @@
             "program": "${env:OutRoot}Winx64/Product/BeSQLite-Gtest/BeSQLiteTest.exe",
             "args": [
                 "--gtest_break_on_failure",
-                "--gtest_filter=BeSQLiteDbTests.MaxBlobSize"
+                "--gtest_filter=BeSQLiteDbTests.UnicodeSupport"
```
Comment (pmconne) [inline .vscode/launch.json:56]:
```suggestion
                "--gtest_filter=BeSQLiteDbTests.MaxBlobSize"
```
Revert.

File: iModelCore/BeSQLite/PublicAPI/BeSQLite/BeSQLite.h [hunk 2]
```diff
@@ -2445,6 +2403,8 @@ struct DbFile : NonCopyableClass
     void SaveCachedProperties(bool isCommit);
     Utf8String GetLastError(DbResult* lastResult) const;
     void SaveCachedBlvs(bool isCommit);
+    DbResult SetNoCaseCollation(NoCaseCollation col) const;
```
Comment (pmconne) [inline iModelCore/BeSQLite/PublicAPI/BeSQLite/BeSQLite.h:2406]: Why is this function `const`?

File: iModelCore/BeSQLite/Tests/NonPublished/BeSQLiteDb_Test.cpp [hunk 3]
```diff
@@ -1696,4 +1696,125 @@ TEST_F (BeSQLiteDbTests, Limits)
     ASSERT_EQ(250000000, m_db.GetLimit(DbLimits::VdbeOp));
     ASSERT_EQ(0, m_db.GetLimit(DbLimits::WorkerThreads));
     m_db.AbandonChanges();
-}
\ No newline at end of file
+}
+//---------------------------------------------------------------------------------------
+// @bsimethod
+//---------------------------------------------------------------------------------------
+#if 0 // Require ICU
```
Comment (pmconne) [inline iModelCore/BeSQLite/Tests/NonPublished/BeSQLiteDb_Test.cpp:1703]: Do you have some plan to enable this test? Does it fail to compile?

File: iModelCore/BeSQLite/PublicAPI/BeSQLite/BeSQLite.h (hunk 2 above)
Comment (khanaffan) [reply iModelCore/BeSQLite/PublicAPI/BeSQLite/BeSQLite.h]: Fixed.

File: iModelCore/BeSQLite/PublicAPI/BeSQLite/BeSQLite.h [hunk 4]
```diff
@@ -2424,6 +2381,7 @@ struct DbFile : NonCopyableClass
     Savepoint m_defaultTxn;
     BeBriefcaseId m_briefcaseId;
     StatementCache m_statements;
+    mutable NoCaseCollation m_noCaseCollation;
```
Comment (pmconne) [inline iModelCore/BeSQLite/PublicAPI/BeSQLite/BeSQLite.h:2384]:
```suggestion
    NoCaseCollation m_noCaseCollation;
```
Now that `SetNoCaseCollation` is no longer `const`.

**Comment by khanaffan** [comment]
@tcobbs-bentley do you have any idea why Android build keep failing with following error for me? 

```
* What went wrong:
Could not determine the dependencies of task ':app:packageDebug'.
> Could not create task ':app:mergeDebugAssets'.
   > Cannot use @TaskAction annotation on method IncrementalTask.taskAction$gradle_core() because interface org.gradle.api.tasks.incremental.IncrementalTaskInputs is not a valid parameter to an action method.
```

**Comment by tcobbs-bentley** [comment]
> @tcobbs-bentley do you have any idea why Android build keep failing with following error for me?
> 
> ```
> * What went wrong:
> Could not determine the dependencies of task ':app:packageDebug'.
> > Could not create task ':app:mergeDebugAssets'.
>    > Cannot use @TaskAction annotation on method IncrementalTask.taskAction$gradle_core() because interface org.gradle.api.tasks.incremental.IncrementalTaskInputs is not a valid parameter to an action method.
> ```

It's failing because after I updated the CI build setup yesterday to use a newer Android SDK to prevent build failures on a new build machine, things are failing. I have been doing everything I can to fix the problems, but every time I fix one problem a new one shows up. My latest fixes are currently running in two linked PRS:

https://github.com/iTwin/imodel-native/pull/738
https://github.com/iTwin/imodel-native-internal/pull/487

**Review by rschili — APPROVED** [review]

**Comment by tcobbs-bentley** [comment]
The PRs to fix the Android build problem have now merged, and I triggered another pipeline run here. Hopefully the pipeline will complete successfully now.

**Pull request merged**